packages/app-dashboard/src/pages/user-dashboard/app-details.tsx

```diff
@@ -2,9 +2,15 @@ import { Helmet } from 'react-helmet';
 import { useReadAuthInfo } from '@/components/consent/hooks/useAuthInfo';
 import UserAuthenticatedConsentForm from '@/components/consent/components/UserAuthenticatedConsentForm';
 import UserHeader from '@/components/layout/UserHeader';
+import { useAuthGuard } from '@/components/user/AuthGuard';
 
 export default function AppDetailsPage() {
   const { authInfo, sessionSigs } = useReadAuthInfo();
+  const authGuardElement = useAuthGuard();
+
+  if (authGuardElement) {
+    return authGuardElement;
```

Comment:
Following the comment above. If the guard is pending, then this page should decide what to do. Almost always showing a `<Loading />` until the guard resolves and redirects or says it finished without issue

This way the guard has also the responsibility of telling the caller what to BLINDLY render.